### PR TITLE
fix(brakes): Instinctive disconnect only when AB active

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #[issue_no]
#9074

## Summary of Changes

Autobrake should only disarm from Instinctive Disconnect button if AB is in progress.
Added handling of MSFS AB DISARM binding input

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Instinctive disconnect (Autothrottle disc button) should only disarm AB when AB is decelerating (Except in RTO)
MSFS AB DISARM input should always set the AB knob in disarm position

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs [hunk 1]
```diff
@@ -229,7 +229,8 @@ impl A380AutobrakeController {
             active_id: context.get_identifier("AUTOBRAKES_ACTIVE".to_owned()),
             rto_mode_armed_id: context.get_identifier("AUTOBRAKES_RTO_ARMED".to_owned()),
 
-            external_disarm_event_id: context.get_identifier("AUTOBRAKE_DISARM".to_owned()),
+            external_disarm_event_id: context
```
Comment (flogross89) [inline fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs:232]:
```suggestion
            external_deactivation_event_id: context
```
and whereevery else this occurs :)

File: fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs [hunk 2]
```diff
@@ -412,10 +413,11 @@ impl A380AutobrakeController {
         // when a simulation is started in flight, some values need to be ignored for a certain time to ensure
         // an unintended disarm is not happening
         (self.deceleration_governor.is_engaged() && self.should_disarm_due_to_pedal_input())
+            || (self.deceleration_governor.is_engaged()
+                && (self.external_disarm_event && self.mode != A380AutobrakeMode::RTO))
```
Comment (flogross89) [inline fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs:417]:
```suggestion
                && (self.external_deactivation_event && self.mode != A380AutobrakeMode::RTO))
```

**Review by flogross89 — CHANGES_REQUESTED** [review]
Cool :)

**Review by flogross89 — APPROVED** [review]

**Comment by BravoMike99** [comment]
Quality Assurance Report

Discord Username : bruno_pt99 (non QA member)
Object of testing : https://github.com/flybywiresim/aircraft/pull/9085
Aircraft : A380X
Tier of Testing : 1
Date : 18/10/2024

Testing Process:
Binded keys to MSFS AUTOTHROTTLE_DISCONNECT & AUTOBRAKE_DISARM.
Performed a RTO, autothrottle disconnect did not deactivate autobrake ✔️ 
Activated ATHR in the air and armed autobrake, pressing autothrottle disconnect bind did not disarm autobrake ✔️ 
Armed autobrake  and pressing the autobrake disarm did set the autobrake knob to disarm position. ✔️ 
During rollout with autobrake active, auto throttle disconnect bind did disconnect the autobrake & knob moved to disarm position ✔️ 
Negatives:

N/A
Testing Results:
Passed

**Pull request merged**